### PR TITLE
Added Guide for plans for Testing in Karpenter

### PR DESCRIPTION
**Pull request by njtran**
This document details plans for making testing within Karpenter more comprehensive, accessible, and configurable.

**4. Does this change impact docs?**
- [X] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/preview/testing-guide
Comment (felix-zhe-huang) [inline website/content/en/preview/testing-guide:3]: This "Users that want to test more ..., resulting in much less testing" sounds a bit redundant. May want to rephrase it. For example.

"Users who want to run more comprehensive tests are limited by the lack of test automation and configurability."

File: website/content/en/preview/testing-guide
Comment (felix-zhe-huang) [inline website/content/en/preview/testing-guide:5]: Is this referring to the CI infrastructure? If so, let's spell it out.

File: website/content/en/preview/testing-guide
Comment (felix-zhe-huang) [inline website/content/en/preview/testing-guide:29]: it is not clear what does it mean by "monitored by the Host Cluster". What is the "Host Cluster"?

File: website/content/en/preview/testing-guide
Comment (felix-zhe-huang) [inline website/content/en/preview/testing-guide:28]: Is this optional?

File: website/content/en/preview/testing-guide
Comment (felix-zhe-huang) [inline website/content/en/preview/testing-guide:33]: Do we repeat the tests ourselves even the contributors may have run tests using similar setup?

File: website/content/en/preview/testing-guide
Comment (njtran) [reply website/content/en/preview/testing-guide:5]: I think it's implied that infrastructure will be CI within the use-cases of the testing detailed below. The design doc will also give more context on this, and what will be CI/CD.

File: website/content/en/preview/testing-guide
Comment (njtran) [reply website/content/en/preview/testing-guide:28]: This will not be optional.
> a maintainer will need

A maintainer will be able to do `/ok-to-test` on their own PRs.

File: website/content/en/preview/testing-guide
Comment (njtran) [reply website/content/en/preview/testing-guide:29]: Cleared up the language a bit. Let me know what you think

File: website/content/en/preview/testing-guide
Comment (njtran) [reply website/content/en/preview/testing-guide:33]: We'll need to ensure that a PR can only be merged after an `/ok-to-test` is successful. I included this as a way for developers to unblock themselves and iterate quicker if maintainers aren't quick enough.

**Review by bwagner5 — APPROVED** [review]
lgtm

**Pull request merged**
